### PR TITLE
Issue 779/outer product with sparse vectors

### LINKED ISSUE #779: Cannot get outer product of a sparse vector and a dense vector
Attempting to get the outer product of a sparse vector and a dense vector by themselves fails. For instance:

```scala
val a = breeze.linalg.DenseVector.zeros[Double](250)
val b = breeze.linalg.SparseVector.zeros[Double](300)
a * b.t
<console>:28: error: could not find implicit value for parameter op: breeze.linalg.operators.OpMulMatrix.Impl2[breeze.linalg.DenseVector[Double],breeze.linalg.Transpose[breeze.linalg.SparseVector[Double]],That]
       a * b.t
```

However, this works perfectly fine when the sparse vector is converted to a dense vector first:

```scala
res14: a * b.toDenseVector.t
breeze.linalg.DenseMatrix[Double] =
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ... (300 total)
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  ...
0.0  0.0  0...
```

### PR DESCRIPTION
**Pull request by bluesheeptoken**
First version of the implementation of outer product between `DenseVector` and `Transposed SparseVector`. Should close #779 

I am very welcome to reviews, I was not sure in which trait I should put the operations implementations.

I tried to mutualize these 2 operations implementations with macro, without success.

### CODE REVIEW

**Comment by bluesheeptoken** [comment]
Hey @dlwh, I submitted this PR for outer product.

I have seen several issues I might have time to handle, however I prefer having a review on this PR before starting another one. To make sure I do not have already wrong habits.

Thanks !

File: math/src/main/scala/breeze/linalg/operators/SparseVectorOps.scala [hunk 1]
```diff
@@ -382,6 +400,25 @@ trait DenseVector_SparseVector_Ops { this: SparseVector.type =>
...
+  implicit def implOpMulMatrix_DVT_SVTt_eq_SMT[@expand.args(Int, Double, Float, Long, Complex) T](
+    implicit @expand.sequence[T](0, 0.0, 0.0f, 0L, Complex.zero) zero: T)
+    : OpMulMatrix.Impl2[DenseVector[T], Transpose[SparseVector[T]], CSCMatrix[T]] = {
+    new OpMulMatrix.Impl2[DenseVector[T], Transpose[SparseVector[T]], CSCMatrix[T]] {
+      def apply(a: DenseVector[T], b: Transpose[SparseVector[T]]): CSCMatrix[T] = {
+        val sizeHint = a.size * b.inner.activeSize
+        val res = new CSCMatrix.Builder[T](a.size, b.inner.size, sizeHint)
+
+        for {(i, aValue) <- a.activeIterator}
+          for {(j, bValue) <- b.inner.activeIterator}
```
Comment (dlwh) [inline math/src/main/scala/breeze/linalg/operators/SparseVectorOps.scala:415]: i think you need to invert these loops to use keysAlreadySorted? CSC is column-major

Comment (bluesheeptoken) [reply math/src/main/scala/breeze/linalg/operators/SparseVectorOps.scala:414]: Completely right

**Comment by bluesheeptoken** [comment]
> Appreciate it! I'm pretty sure there's a bug? Could you test for num-non-zero >2 and length > 2?

My pleasure !

I completed the tests, which bug did you have in mind ?

**Comment by dlwh** [comment]
thanks again!

**Pull request merged**